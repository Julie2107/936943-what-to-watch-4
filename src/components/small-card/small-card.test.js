import React from "react";
import renderer from "react-test-renderer";
import MovieSmallCard from "./small-card.jsx";
import {Router} from "react-router-dom";
import history from "../../history";

const movie = {
  title: `Pulp Fiction`,
  poster: `http://placekitten.com/245/175`,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
};

it(`Render moviesmallcard`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <MovieSmallCard
            movie={movie}
            onTitleClick={()=>{}}
            onMouseHover={()=>{}}
            onMouseHoverOff={()=>{}}
            isPlaying={false}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
