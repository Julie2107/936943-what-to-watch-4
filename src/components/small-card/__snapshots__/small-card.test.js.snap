// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render moviesmallcard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <video
      className="player__video"
      muted={true}
      poster="http://placekitten.com/245/175"
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/2"
      onClick={[Function]}
    >
      Pulp Fiction
    </a>
  </h3>
</article>
`;
